### PR TITLE
Rename bp-table -> pt-table. Upgrade script updates

### PR DESCRIPTION
**Pull request by themadcreator**
Add rename for all classname strings `bp-table` -> `pt-table`
Change `onFocus` from rename to warning since it's likely to clobber common uses
Add other core renames to script

### CODE REVIEW

File: packages/docs-app/src/styles/_sections.scss [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
 // CSS examples
 
-#{reference("pt-file-upload")},
+#{reference("pt-file-input")},
```
Comment (tgreenwatts) [inline packages/docs-app/src/styles/_sections.scss]: Is this one missed from https://github.com/palantir/blueprint/pull/1880 ?

Comment (themadcreator) [reply packages/docs-app/src/styles/_sections.scss]: Looks like it. Good thing we have a script!

**Comment by cmslewis** [comment]
@themadcreator @adidahiya @giladgray @llorca 

I really don't like that `pt-table` is a totally separate component from `pt-table-X`. It's innately confusing, and it's such a weird exception to our library-wide convention that every `pt-[component]-X` class is a child, variant, or modifier of `pt-[component]`.

Ideas:
- Rename `<Table>` to `<SomethingElse>` (e.g. `<DataGrid>`).
- Introduce a top-level `pt-tablejs` or `pt-table-js` class, and change all `packages/table` CSS classes to use this prefix.
- Rename the existing `pt-table` to `pt-table-html` (consistent with the "Table (HTML)" title).
- Do nothing; live with the awkwardness.

File: packages/core/src/common/_color-aliases.scss [hunk 2]
```diff
@@ -7,7 +7,7 @@ Color aliases
 These variables are semantic aliases of our [colors](#colors). They are used throughout Blueprint
 itself to ensure consistent color usage across components.
 
-<table class=pt-table>
+<table class=pt-html-table>
```
Comment (giladgray) [inline packages/core/src/common/_color-aliases.scss]: missing quotes. though it seems to work.

File: packages/core/src/common/classes.ts [hunk 3]
```diff
@@ -156,7 +156,7 @@ export const TAB_LIST = "pt-tab-list";
 export const TAB_PANEL = "pt-tab-panel";
 export const TABS = "pt-tabs";
 
-export const TABLE = "pt-table";
+export const TABLE = "pt-html-table";
```
Comment (giladgray) [inline packages/core/src/common/classes.ts]: should we rename constant too, to match?

File: scripts/upgrade-blueprint-2.0.0-rename [hunk 4]
```diff
@@ -99,42 +94,103 @@ function rename() {
...
+    warn $1 $2 "$PREFIX\\b$1\\b"
+}
+
+function renamePartialClass() {
+    # Don't add word boundary so that partial css classnames rename
+    rename $1 $2 "\\b$1"
+}
+
+function renameString() {
+    # Don't add word boundary so that partial css classnames rename
```
Comment (giladgray) [inline scripts/upgrade-blueprint-2.0.0-rename]: invalid comment

File: scripts/upgrade-blueprint-2.0.0-rename [hunk 5]
```diff
@@ -99,42 +94,103 @@ function rename() {
...
+    # Add prefix and word boundaries to search string
+    rename $1 $2 "$PREFIX\\b$1\\b"
+}
+
+function warnProp() {
+    warn $1 $2 "$PREFIX\\b$1\\b"
+}
+
+function renamePartialClass() {
+    # Don't add word boundary so that partial css classnames rename
```
Comment (giladgray) [inline scripts/upgrade-blueprint-2.0.0-rename]: invalid comment?

File: packages/core/src/common/_color-aliases.scss (hunk 2 above)
Comment (llorca) [reply packages/core/src/common/_color-aliases.scss]: def works. let's add them for good measure

File: scripts/upgrade-blueprint-2.0.0-rename (hunk 5 above)
Comment (themadcreator) [reply scripts/upgrade-blueprint-2.0.0-rename]: nope, this is still relevant, notice how we don't add the the `\\b` to this one.

File: scripts/upgrade-blueprint-2.0.0-rename (hunk 4 above)
Comment (themadcreator) [reply scripts/upgrade-blueprint-2.0.0-rename]: oh yeah this is invalid

File: packages/table/src/table.tsx [hunk 6]
```diff
@@ -785,12 +785,12 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
...
                     quadrantRef={this.refHandlers.mainQuadrant}
                     ref={this.refHandlers.quadrantStack}
-                    renderMenu={this.renderMenu}
+                    menuRenderer={this.renderMenu}
                     rowHeaderCellRenderer={this.renderRowHeader}
                     rowHeaderRef={this.refHandlers.rowHeader}
                     scrollContainerRef={this.refHandlers.scrollContainer}
                 />
-                <div className={classNames(Classes.TABLE_OVERLAY_LAYER, "bp-table-reordering-cursor-overlay")} />
+                <div className={classNames(Classes.TABLE_OVERLAY_LAYER, "pt-table-reordering-cursor-overlay")} />
```
Comment (adidahiya) [inline packages/table/src/table.tsx]: nit: we should make this a constant in `Classes`

**Pull request merged**